src/Apps/Order/routes.tsx

```diff
@@ -65,9 +103,22 @@ export const routes: RouteConfig[] = [
         }
         order: ecommerceOrder(id: $orderID) {
           state
+          requestedFulfillment {
+            __typename
+          }
+          creditCard {
+            id
+          }
         }
       }
     `,
+    render: ({ Component, props }) => {
+      if (Component && props) {
+        if (!shouldRedirect(props)) {
```

Comment:
Extracted this out as a function-- I didn't move it to its own file since it didn't feel too unwieldy but that's an easy thing to follow up on.